cpp/src/fil/common.cuh

```diff
@@ -145,9 +159,13 @@ struct predict_params {
   int num_cols;
   algo_t algo;
   int max_items;  // only set and used by infer()
-  int num_output_classes;
-  // so far, only 1 or 2 is supported, and only used to output probabilities
-  // from classifier models
+  // number of outputs for the forest for each data instance (sample) (row)
+  int num_outputs;
+  // for class probabilities, this is the number of classes considered
+  // ignored otherwise
+  int num_classes;
```

Comment:
needed for shared memory allocation for votes, which may be different from num_outputs (which is used in e.g. transform_k)